client/tools/rhncfg/config_management/rhncfg-manager.8
.\" This manpage has been automatically generated by docbook2man 
.\" from a DocBook document.  This tool can be found at:
.\" <http://shell.ipoline.com/~elmert/comp/docbook2X/> 
.\" Please send any bug reports, improvements, comments, patches, 
.\" etc. to Steve Cheng <steve@ggi-project.org>.
.TH "RHNCFG-MANAGER" "8" "03 August 2005" "Version 4.0" ""

.SH NAME
rhncfg-manager \- a tool used managing RHN's central repository of configuration files  and channels.
.SH SYNOPSIS

.nf
    

\fBrhncfg-manager\fR [ \fBMODE\fR ] [ \fB--server-name \fIname\fB\fR ] [ \fBparams\fR ]

.fi
.SH "DESCRIPTION"
.PP
The RHN Configuration Manager (\fBrhncfg-manager\fR) is used to
manage Spacewalk configuration files and channels.
.PP
This tool offers a command line alternative to the configuration management
features found within the RHN website (hosted or Red Hat Satellite), as well
as the ability to script some or all of the related maintenance.
.PP
This tool is intended for use by the Config Administrator and therefore
requires an RHN username and password that has the appropriate permission
set. (The username may be specified in
\fI/etc/sysconfig/rhn/rhncfg-manager.conf\fR or in the [rhncfg-manager] section
of \fI~/.rhncfgrc\fR).
.PP
Note that when run as root, rhncfg-manager attempts to pull in needed
configuration values from the Red Hat Update Agent (up2date). When run as
something other than root, configuration changes may be needed within the
\fI~/.rhncfgrc\fR file. The session file is cached in
\fI~/.rhncfg-manager-session\fR to prevent logging in for
every command.
.PP
The \fI~/.rhncfgrc\fR file overrides the values set in
\fI/etc/sysconfig/rhn/rhncfg-manager.conf\fR which, by
default, uses the values set in
\fI/etc/sysconfig/rhn/up2date\fR\&. Please review
\fIrhncfg-manager.conf\fR for further details.
.PP
The Spacewalk Configuration Manager offers these primary modes:
\fBadd\fR, \fBcreate-channel\fR,
\fBdiff\fR, \fBdiff-revisions\fR,
\fBdownload-channel\fR, \fBget\fR,
\fBlist\fR, \fBlist-channels\fR,
\fBremove\fR, \fBremove-channel\fR,
\fBrevisions\fR, \fBupdate\fR,
\fBupload-channel\fR\&.
.PP
Context sensitive help is available for each "mode", e.g.,
rhncfg-manager diff-revisions --help
.PP

This tool utilizes a "cvs style" command syntax: command mode options

For example: rhncfg-manager diff-revisions --help
.SH "MODES"
.TP
\fBadd [ \fIOPTIONS\fB ] \fIFILE\fB [ \fIFILE\fB ... ]\fR
Add a file to a config channel.

-c\fICHANNEL_LABEL\fR, --channel=\fICHANNEL_LABEL\fR -- upload file(s) in this config channel

-d\fIDEST_FILE\fR, --dest-file=\fIDEST_FILE\fR -- upload the file as this path

-t\fITOPDIR\fR, --topdir=\fITOPDIR\fR -- make all files relative to this string

--delim-start=\fIDELIM_START\fR -- start delimiter for variable interpolation

--delim-end=\fIDELIM_END\fR -- end delimiter for variable interpolation
.TP
\fBcreate-channel \fICHANNEL_LABEL\fB [ \fICHANNEL_LABEL ...\fB ]\fR
Create a config channel for your organization.
.TP
\fBdiff [ \fIOPTIONS\fB ] \fIFILE\fB [ \fIFILE\fB ... ]\fR
Differentiating between latest configuration files in a channel
and whats on disk.

-c\fICHANNEL_LABEL\fR, --channel=\fICHANNEL_LABEL\fR -- get file(s) from this config channel

-r\fIREVISION\fR, --revision=\fIREVISION\fR -- use this revision

-d\fIDEST_FILE\fR, --dest-file=\fIDEST_FILE\fR -- upload the file as this path

-t\fITOPDIR\fR, --topdir=\fITOPDIR\fR -- make all files relative to this string
.TP
\fBdiff-revisions [ \fIOPTIONS\fB ] \fIFILE\fB\fR
Compare different versions of a file across channels and
revisions. See documentation for further details.

-c\fICHANNEL_LABEL\fR, --channel=\fICHANNEL_LABEL\fR -- use this config channel

-r\fIREVISION\fR, --revision=\fIREVISION\fR -- use this revision
.TP
\fBdownload-channel (-t|--topdir=)\fITOP_LEVEL_DIRECTORY\fB [ \fIOPTIONS\fB ] \fICHANNEL_LABEL\fB [ \fICHANNEL_LABEL\fB ... ]\fR
Download all the configuration files in a channel to disk.

-t\fITOPDIR\fR, --topdir=\fITOPDIR\fR -- directory all the file paths are relative to
.TP
\fBget [ \fIOPTIONS\fB ] \fIFILE\fB [ \fIFILE\fB ... ]\fR
Get the contents of a particular configuration file. Outputs
to stdout.

-c\fICHANNEL_LABEL\fR, --channel=\fICHANNEL_LABEL\fR -- get file(s) from this config channel

-t\fITOPDIR\fR, --topdir=\fITOPDIR\fR -- make all files relative to this string

-r\fIREVISION\fR, --revision=\fIREVISION\fR -- get this file revision
.TP
\fBlist \fICHANNEL_LABEL\fB\fR
List all configuration files in a channel.
.TP
\fBlist-channels\fR
List all configuration channels.
.TP
\fBremove [ \fIOPTIONS\fB ] \fIFILE\fB [ \fIFILE\fB ... ]\fR
Remove a configuration file from a channel. May be prompted for
a username and password.

-c\fICHANNEL_LABEL\fR, --channel=\fICHANNEL_LABEL\fR -- remove file(s) from this config channel

-t\fITOPDIR\fR, --topdir=\fITOPDIR\fR -- make all files relative to this string
.TP
\fBremove-channel \fICHANNEL_LABEL\fB [ \fICHANNEL_LABEL\fB ... ]\fR
Remove a configuration channel. May be prompted for
a username and password.
.TP
\fBrevisions [ \fIOPTIONS\fB ] \fIFILE\fB [ \fIFILE\fB ... ]\fR
Determine the number of configuration file revisions.

-c\fICHANNEL_LABEL\fR, --channel=\fICHANNEL_LABEL\fR -- use this config channel
.TP
\fBupdate [ \fIOPTIONS\fB ] \fIFILE\fB [ \fIFILE\fB ... ]\fR
Create a new revision of a file in a channel (or add the first
revisions to that channel if none existed before for the given
path).

-c\fICHANNEL_LABEL\fR, --channel=\fICHANNEL_LABEL\fR -- upload file(s) in this config channel

-d\fIDEST_FILE\fR, --dest-file=\fIDEST_FILE\fR -- upload the file as this path

-t\fITOPDIR\fR, --topdir=\fITOPDIR\fR -- make all files relative to this string

--delim-start=\fIDELIM_START\fR -- start delimiter for variable interpolation

--delim-end=\fIDELIM_END\fR -- end delimiter for variable interpolation
.TP
\fBupload-channel [ \fIOPTIONS\fB ] \fICHANNEL_LABEL\fB [ \fICHANNEL_LABEL\fB ... ]\fR
Upload multiple configuration files (a tree) to a channel from
disk.

-t\fITOPDIR\fR, --topdir=\fITOPDIR\fR -- directory all file paths are relative to
.TP
\fB-h, --help\fR
Display the help screen with a list of options.

Use context sensitive help as well with the diffent "modes".
E.g., rhncfg-manager upload-channel --help
.SH "SEE ALSO"

rhncfg-client (8)
.SH "AUTHORS"

Mihai Ibanescu <misa@redhat.com>

Bret McMillan <bretm@redhat.com>

Todd Warner <taw@redhat.com> (man page only)


client/tools/rhncfg/config_management/rhncfg-manager.py
#!/usr/bin/python
#
# Copyright (c) 2008 Red Hat, Inc.
#
# This software is licensed to you under the GNU General Public License,
# version 2 (GPLv2). There is NO WARRANTY for this software, express or
# implied, including the implied warranties of MERCHANTABILITY or FITNESS
# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
# along with this software; if not, see
# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
# 
# Red Hat trademarks are not licensed under GPLv2. No permission is
# granted to use or replicate Red Hat trademarks that are incorporated
# in this software or its documentation. 
#

RHNROOT = '/usr/share/rhn'
import sys
if RHNROOT not in sys.path:
    sys.path.append(RHNROOT)

from config_common.rhn_main import BaseMain

class Main(BaseMain):
    modes = [
        'add',
        'create-channel',
        'diff',
        'diff-revisions',
        'download-channel',
        'get',
        'list',
        'list-channels',
        'remove',
        'remove-channel',
        'revisions',
        'update',
        'upload-channel',
    ]
    plugins_dir = 'config_management'
    config_section = 'rhncfg-manager'
    mode_prefix = 'rhncfg'

if __name__ == '__main__':
    try:
        sys.exit(Main().main() or 0)
    except KeyboardInterrupt:
        sys.stderr.write("user interrupted\n")
        sys.exit(0)

